### DISCUSSION TITLE
Has Valkey been enabled as an allowed engine for ElastiCache in CDK?

### DISCUSSION THREAD

**Discussion by jaydm26** [aws/aws-cdk#31884] [Q&A]
CDK Docs (https://docs.aws.amazon.com/cdk/api/v2/python/aws_cdk.aws_elasticache/CfnCacheCluster.html) don't show Valkey as an option for ElastiCache engine. However, CFN Docs suggest that valkey is an allowed value (https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-elasticache-cachecluster.html#cfn-elasticache-cachecluster-engine).

Thus, I am wondering if we can now create ElastiCache Cluster with a valkey engine with CDK or not.

Thank you in advance for your help!

**Comment by khushail** [comment]
Hi @jaydm26 , you can create ElasticCache cluster with valkey engine using L1 construct - CfnCacheCluster as there are no official L2 construct from CDK providing this support.

**Comment by andremantaswow** [comment]
any updates on this?

**Comment by cbarreholm** [comment]
https://docs.aws.amazon.com/cdk/api/v2/python/aws_cdk.aws_elasticache/CfnCacheCluster.html currently list `valkey`

> `engine (str) – The name of the cache engine to be used for this cluster. Valid values for this parameter are: memcached | valkey | redis`

Tried that value in a CDK diff 

```
Resources
[~] AWS::ElastiCache::CacheCluster MyCache MyCache replace
 └─ [~] Engine (requires replacement)
     ├─ [-] redis
     └─ [+] valkey
```
